src/spi.rs

```diff
@@ -551,6 +551,31 @@ macro_rules! spi {
                         self.spi.sr.read().ovr().is_overrun()
                     }
 
+                    /// Clears the MODF flag, which indicates that a
+                    /// mode fault has occurred.
+                    pub fn clear_modf(&mut self) {
+                        self.spi.ifcr.write(|w| w.modfc().clear());
+                    }
+
+                    /// Disables the SPI peripheral. Any SPI operation is
+                    /// stopped and disabled, the internal state machine is
+                    /// reset, all the FIFOs content is flushed, the MODF
+                    /// flag is cleared, the SSI flag is cleared, and the
+                    /// CRC calculation is re-initialized. Clocks are not
+                    /// disabled.
+                    pub fn disable(&mut self) {
```

Comment:
I'm assuming that this puts the peripheral in a state where it can no longer send data since it's disabled? If that is the case with the system proposed here it would be easily possible to create an SPI device that can be given to a driver but not actually used by since the peripheral itself is disabled, I don't think that is a particularly good idea.

If that's the case the correct thing to do would be to represent this in the type system IMHO, for example via a type state transition or similar.